projects/pgai/pgai/vectorizer/vectorizer.py

```diff
@@ -1177,62 +1523,116 @@ async def _generate_embeddings(
         import numpy as np
 
         records_without_embeddings: list[EmbeddingRecord] = []
-        loading_errors: list[tuple[SourceRow, LoadingError]] = []
+        errors: list[tuple[SourceRow, ProcessingError]] = []
         documents: list[str] = []
-        for item in items:
-            pk_values = self._get_item_pk_values(item)
-            try:
-                payload = self.vectorizer.config.loading.load(item)
-            except Exception as e:
-                if self.features.loading_retries:
-                    loading_errors.append((item, (LoadingError(e=e))))
-                continue
 
-            payload = self.vectorizer.config.parsing.parse(item, payload)
-            chunks = self.vectorizer.config.chunking.into_chunks(item, payload)
-            for chunk_id, chunk in enumerate(chunks, 0):
-                formatted = self.vectorizer.config.formatting.format(chunk, item)
-                records_without_embeddings.append(pk_values + [chunk_id, formatted])
-                documents.append(formatted)
-
-        if loading_errors:
-            yield [], loading_errors
+        async with timed_debug_log(f"Pre-processed {len(items)} items"):
+            for item in items:
+                pk_values = self._get_item_pk_values(item)
+                try:
+                    payload = self.vectorizer.config.loading.load(item)
+                except Exception as e:
+                    if self.features.loading_retries:
+                        errors.append((item, LoadingError(e=e)))
+                    continue
+
+                try:
+                    payload = self.vectorizer.config.parsing.parse(item, payload)
+                except Exception as e:
+                    if self.features.loading_retries:
+                        errors.append((item, ParsingError(e=e)))
+                        continue
+                    else:
+                        raise e
+                chunks = self.vectorizer.config.chunking.into_chunks(item, payload)
+                try:
+                    for chunk_id, chunk in enumerate(chunks, 0):
+                        formatted = self.vectorizer.config.formatting.format(
+                            chunk, item
+                        )
+                        records_without_embeddings.append(
+                            pk_values + [chunk_id, formatted]
+                        )
+                        documents.append(formatted)
+                except Exception as e:
+                    if self.features.loading_retries:
+                        errors.append((item, FormattingError(e=e)))
+                        continue
+                    else:
+                        raise e
```

Comment:
I somehow thought the goal still was to do this all within the original transaction to take exactly enough to fill an embedding batch? Rn we still use the configured batch size, are you planning to do a separate PR for this or are just not doing it for now?